com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -308,23 +311,23 @@ public void NetworkHide(ulong clientId)
         /// <param name="clientId">The client to hide the objects from</param>
         public static void NetworkHide(List<NetworkObject> networkObjects, ulong clientId)
         {
-            if (!NetworkManager.Singleton.IsServer)
+            if (networkObjects == null || networkObjects.Count == 0)
             {
-                throw new NotServerException("Only server can change visibility");
+                throw new ArgumentNullException("At least one NetworkObject has to be provided");
             }
 
-            if (clientId == NetworkManager.Singleton.ServerClientId)
+            NetworkManager networkManager = networkObjects[0].NetworkManager;
```

Comment:
Same comment as above, can this not be (down in the loop):

```suggestion
            NetworkManager networkManager = networkObjects[i].NetworkManager;
```

Reply:
Nope, because of how things gets packed. What we do is:
1. Get the first NetworkManager
2. Iterate all NetworkObjects in a safety-check loop to ensure they are all spawned and such. We also check that they all belong to the NetworManager we previouly grabbed.
3. Send off the packet